src/main.c

```diff
@@ -472,10 +462,43 @@ static void next_page(fz_context *ctx, ui_state *ui)
 {
   synctex_set_target(send(synctex, ui->eng, NULL), 0, NULL, 0);
   ui->page += 1;
-  pan_to_top(ctx, ui);
+  // FIXME: Same remark as in previous_page.
+  pan_to(ctx, ui, PAN_TO_TOP);
   schedule_event(RELOAD_EVENT);
 }
 
+static void ui_pan(fz_context *ctx, ui_state *ui, float factor)
+{
+  fz_point scale = get_scale_factor(ui->window);
+
+  txp_renderer_config *config = txp_renderer_get_config(ctx, ui->doc_renderer);
+
+  txp_renderer_bounds bounds;
+  if (!txp_renderer_page_bounds(ctx, ui->doc_renderer, &bounds))
+    return;
+
+  float delta = bounds.window_size.y * scale.y * factor;
+  float range = bounds.pan_interval.y < 0 ? 0 : bounds.pan_interval.y;
+
+  fprintf(stderr, "ui_pan: factor:%.02f delta:%.02f current:%.02f range:%.02f\n",
+          factor, delta, config->pan.y, range);
```

Comment:
Did you mean to leave the unconditional printing here?

(Note: I failed to find where the printing output goes when I run `texpresso` from emacs, so in the end I never used it.)

Reply:
It is accidental, thanks for catching that :).

The output goes into `*texpresso-stderr*` buffer, which is truncated to keep only the last 50k of log.